chatter.js

```diff
@@ -1,12 +1,80 @@
 Chatter = {
-    options: {
-        messageLimit: 30,
-        nickProperty: "username",
+  options: {
+    messageLimit: 30,
+    nickProperty: "username",
+  }
+};
+
+Chatter.configure = function (opts) {
+  _.extend(this.options, opts);
+};
+
+/**
+ * @summary Adds user to Chatter and defines its privileges
+ * @locus Server
+ * @param {string} userId Unique string identifying the users of your application.
+ * @param {string} userType Defines the restricitons on the new user. Can either be set to "admin" or "standard".
+ * @returns {string} userId
+ */
+Chatter.addUser = function(userId, userType) {
+   check(userId, String);
+   check(userType, Match.Maybe(String));
+  const user = Meteor.users.findOne({_id: userId});
+
+  Chatter.User.upsert({
+    userId: userId
+  }, {
+    $set: {
+      userType: userType,
+      nickname: getNickname(user)
     }
+  });
+  return Chatter.User.findOne({userId: userId})._id;
 };
 
+/**
+ * @summary Adds empty room to Chatter.
+ * @locus Server
+ * @param {string} rommName
+ * @returns {string} roomId
+ */
+Chatter.addRoom = function(roomName) {
+  check(roomName, String);
 
-Chatter.configure = function (opts) {
-    _.extend(this.options, opts);
+  return Chatter.Room.insert({name: roomName});
 };
 
+/**
+ * @summary Adds user to room.
+ * @locus Server
+ * @param {string} userId Unique string identifying user.
+ * @param {string} roomName Unique string identifying the room.
+ * @returns {string} userRoomId
+ */
+Chatter.addUserToRoom = function(userId, roomId) {
+  check(userId, String);
+  check(roomId, String);
+  const chatterUserId = Chatter.User.findOne({userId: userId})._id;
```

Comment:
Yes, but they could be providing a valid userId that does not have an associated ChatterUser, In this case, the check will pass and this line will throw a silly error. If you first try to find the chatterUser, check that for existence, then get the chatterUserId, you can give the user a useful error ("UserId has no associated ChatterId") instead  of a nonsense property error on undefined/null